src/components/GeneralInfo/TextInputModal/TextInputModal.test.js

```diff
@@ -1,48 +1,39 @@
+import '@testing-library/jest-dom';
 import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import React from 'react';
 import TextInputModal from './TextInputModal';
 
 describe('TextInputModal', () => {
-  describe('getDerivedStateFromProps', () => {
-    it('should set state value to undefined', () => {
-      expect(
-        TextInputModal.getDerivedStateFromProps(
-          {
-            isOpen: false,
-            value: 'some-value',
-          },
-          { value: 'test' }
-        )
-      ).toEqual({
-        value: undefined,
+  describe('State management with props', () => {
+    it('should reset value to undefined when modal is closed', () => {
+      const { rerender } = render(<TextInputModal isOpen={true} value="foo" />);
+      rerender(<TextInputModal isOpen={false} value="foo" />);
+
+      const input = screen.queryByRole('textbox', {
+        name: /input text/i,
       });
+      expect(input).not.toBeInTheDocument();
```

Comment:
This only asserts that the modal is not in the DOM, though the test description says that it tests if the value is reset to undefined when the modal is closed